### DISCUSSION TITLE
Linux Problem with Asymetric routing and StrongSwan VPN -- GRE Tunnel.

### DISCUSSION THREAD

**Discussion by jylandry** [strongswan/strongswan#845] [General]
I am far from certain the problem is related to Strongswan.  But if someone has an Idea, I will take it.  **I have a single scenario that create an issues and seems related to asymmetric routing over IPSEC VPN/GRE TUNNEL.**

My WAN network is base on a Hub and Spoke topology.  The HUB is a Linux Machine (Ubuntu 18.04) running Strongswan 5.9.4.  It is the only machine that is not behind a NAT gateway (And is the main router for [Quagga] OSPF area 0)

The relevant Spokes are Ubiquiti EdgeRouter (Linux StrongSwan U5.6.3/K4.14.54-UBNT).  Inconsequential for this concern, but other machines are connected to the StrongSwan IKEV2 network, including others EdgeRouters, Linux Machines and some Android phones using Android StronSwan VPN.

GRE Tunnel are riding the IPSEC VPN.  OSPF does maintain reachability.  Beside this single issue, the network is stable and work well.

![text2580](https://user-images.githubusercontent.com/58461233/149032679-573a6669-70ee-49bd-af0c-52528db79fa9.png)

**Problem Scenario : Site CR(Area 111)  has two EdgeRouter in redundancy, 2 VPN Tunnels riding 2 independents ISP line**.  

Under some circumstance, the network does operate in Asymmetric routing mode.  It does create an issues at The HUB.

- On Linux HUB,   pinging a device at SITE CR that result in an Asymmetric routing always timeout.
- On Linux HUB --  tcpdump -i gre-netCR1 --> See the outgoing Echo request.
- On Linux HUB -- tcpdump -i gre-netCR2 --> See the incoming Echo reply.
- On Linux HUB -- The ICMP message is never delivered to the program and the ping timeout
- With asymmetric routing, the same behaviour exist for all TCP protocol including http(s):, ssh.  Furthermore, the behaviour can be observed when the ping/https/ssh originated from any spoke site and is merely routed by the HUB.
- When the routing between HUB and SITE CR uses a single VPN/GRE Line, it works flawlessly.
- Meanwhile, any devices at SITE CR can ping anywhere on the network, even if it generates asymmetric routing between site CR and the HUB.

**Comment by Thermi** [comment]
That seems like an issue with your OSPF where routes are not negotiated in a way that results in per site symmetrical routes.
